game.py

```diff
@@ -13,10 +14,11 @@ def self_play_game(model,
                    verbose=False):
     """
     Plays a game (defined by the env), where a model with MCTS action distribution improvement plays
-    itself. Returns a tuple of (states, winner_vector, action_distributions)
+    itself. Returns a tuple of (states, winner_vector, action_distributions, last_state)
```

Comment:
Why are we including the last_state?

Reply:
It's not currently used anywhere I don't think, but it's not included in the states vector so I thought it would be nice to have a complete game history if desired. Should we remove it?

Reply:
We're gonna use this for training data, which doesn't need last_states right?  I think we should remove it for that reason

Reply:
poof